src/case.rs

```diff
@@ -0,0 +1,146 @@
+use inflector;
+
+/// Enum for converting one case into another. Uses the [`Inflector`][1] crate.
+///
+/// [1]: https://crates.io/crates/Inflector
+#[derive(PartialEq, Debug)]
+pub enum Case {
+  /// A case identity. When converting it makes sure to always return the
+  /// original value.
+  Same,
+  /// Converts to camelCase, like JavaScript.
+  Camel,
+  /// Converts to kebab-case, like CSS.
+  Kebab,
+  /// Converts to snake_case, like Rust.
+  Snake,
+  /// Converts to ClassCase, like Java.
+  Class,
+  /// Converts to SCREAMING_CASE, like environment variables.
+  Screaming,
+  /// Converts to “Sentence case.”
+  Sentence,
+  /// Converts to “Title Case.”
+  Title,
+  /// Converts to “UPPER CASE”.
+  Upper,
+  /// Converts to “lower case”.
+  Lower
+}
+
+// Release the case variants!
+pub use self::Case::*;
+
+impl Case {
+  /// Gets the enum case variant from its lower case name.
+  ///
+  /// # Example
+  /// ```rust
+  /// use ardite::case::*;
+  ///
+  /// assert_eq!(Case::from_name("same").unwrap(), Same);
+  /// assert_eq!(Case::from_name("camel").unwrap(), Camel);
+  /// assert_eq!(Case::from_name("kebab").unwrap(), Kebab);
+  /// assert_eq!(Case::from_name("snake").unwrap(), Snake);
+  /// assert_eq!(Case::from_name("class").unwrap(), Class);
+  /// assert_eq!(Case::from_name("screaming").unwrap(), Screaming);
+  /// assert_eq!(Case::from_name("sentence").unwrap(), Sentence);
+  /// assert_eq!(Case::from_name("title").unwrap(), Title);
+  /// assert_eq!(Case::from_name("upper").unwrap(), Upper);
+  /// assert_eq!(Case::from_name("lower").unwrap(), Lower);
+  /// ```
+  pub fn from_name(string: &str) -> Option<Self> {
```

Comment:
I know, but lower case is provided by the `Inflector` module, so I just use that implementation 😐

He might be using `to_lowercase` internally.